### PR TITLE
chore: add sig config for ubuntu minimal

### PR DESCRIPTION
**Pull request by anujmaheshwari1**
**What type of PR is this?**
This PR adds sig config for Ubuntu Minimal VHD.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: pkg/agent/datamodel/sig_config.go [hunk 1]
```diff
@@ -447,6 +452,20 @@ var (
 		Version:       LinuxSIGImageVersion,
 	}
 
+	SIGUbuntuMinimalContainerd2204ImageConfigTemplate = SigImageConfigTemplate{
+		ResourceGroup: AKSUbuntuResourceGroup,
+		Gallery:       AKSUbuntuGalleryName,
+		Definition:    "2204minimalcontainerd",
+		Version:       "202306.30.0",
```
Comment (cameronmeissner) [inline pkg/agent/datamodel/sig_config.go:459]: can we have a separate const for minimal versions?

File: pkg/agent/datamodel/sig_config.go [hunk 2]
```diff
@@ -447,6 +452,20 @@ var (
 		Version:       LinuxSIGImageVersion,
 	}
 
+	SIGUbuntuMinimalContainerd2204ImageConfigTemplate = SigImageConfigTemplate{
+		ResourceGroup: AKSUbuntuResourceGroup,
+		Gallery:       AKSUbuntuGalleryName,
+		Definition:    "2204minimalcontainerd",
```
Comment (cameronmeissner) [inline pkg/agent/datamodel/sig_config.go:458]: this reminds me, did you make sure to update the set SKU_NAME logic in the vhd builds?

File: pkg/agent/datamodel/sig_config.go (hunk 1 above)
Comment (anujmaheshwari1) [reply pkg/agent/datamodel/sig_config.go:459]: I had thought so but its gonna be testing only after which the version should be same as linuxSIGimage version, hence I did not bother creating a different json file.

File: pkg/agent/datamodel/sig_config.go (hunk 2 above)
Comment (anujmaheshwari1) [reply pkg/agent/datamodel/sig_config.go:458]: I did :) https://github.com/Azure/AgentBaker/pull/3370/files#r1254790224

File: pkg/agent/datamodel/sig_config.go (hunk 1 above)
Comment (cameronmeissner) [reply pkg/agent/datamodel/sig_config.go:459]: makes sense, though I was mainly just talking about a const in the go file here, no need to deal with the json embedding - either way is fine tho

File: pkg/agent/datamodel/sig_config.go (hunk 2 above)
Comment (cameronmeissner) [reply pkg/agent/datamodel/sig_config.go:458]: noice I missed that lol

File: pkg/agent/datamodel/sig_config.go (hunk 1 above)
Comment (anujmaheshwari1) [reply pkg/agent/datamodel/sig_config.go:459]: will let it stay as is if it isnt blocking. will ofcourse revisit once its actively being used/built.

**Review by cameronmeissner — APPROVED** [review]

**Pull request merged**